.github/scripts/Build-Windows.ps1

```diff
@@ -0,0 +1,139 @@
+[CmdletBinding()]
+param(
+    [ValidateSet('x64')]
+    [string] $Target = 'x64',
+    [ValidateSet('Debug', 'RelWithDebInfo', 'Release', 'MinSizeRel')]
+    [string] $Configuration = 'RelWithDebInfo',
+    [switch] $SkipAll,
+    [switch] $SkipBuild,
+    [switch] $SkipDeps
+)
+
+$ErrorActionPreference = 'Stop'
+
+if ( $DebugPreference -eq 'Continue' ) {
+    $VerbosePreference = 'Continue'
+    $InformationPreference = 'Continue'
+}
+
+if ( ! ( [System.Environment]::Is64BitOperatingSystem ) ) {
+    throw "obs-studio requires a 64-bit system to build and run."
+}
+
+if ( $PSVersionTable.PSVersion -lt '7.2.0' ) {
+    Write-Warning 'The obs-studio PowerShell build script requires PowerShell Core 7. Install or upgrade your PowerShell version: https://aka.ms/pscore6'
+    exit 2
+}
+
+function Build {
+    trap {
+        Pop-Location -Stack BuildTemp -ErrorAction 'SilentlyContinue'
+        Write-Error $_
+        Log-Group
+        exit 2
+    }
+
+    $ScriptHome = $PSScriptRoot
+    $ProjectRoot = Resolve-Path -Path "$PSScriptRoot/../.."
+    $BuildSpecFile = "${ProjectRoot}/buildspec.json"
+
+    $UtilityFunctions = Get-ChildItem -Path $PSScriptRoot/utils.pwsh/*.ps1 -Recurse
+
+    foreach($Utility in $UtilityFunctions) {
+        Write-Debug "Loading $($Utility.FullName)"
+        . $Utility.FullName
+    }
+
+    $BuildSpec = Get-Content -Path ${BuildSpecFile} -Raw | ConvertFrom-Json
+
+    if ( ! $SkipDeps ) {
+        Install-BuildDependencies -WingetFile "${ScriptHome}/.Wingetfile"
+    }
+
+    Push-Location -Stack BuildTemp
+    if ( ! ( ( $SkipAll ) -or ( $SkipBuild ) ) ) {
+        Ensure-Location $ProjectRoot
+
+        $CMakeArgs = @()
+        $CmakeBuildArgs = @('--build')
+        $CmakeInstallArgs = @()
+
+        $Preset = "windows-$(if ( $env:CI -ne $null ) { 'ci-' })${Target}"
+
+        $CmakeArgs = @(
+            '--preset', $Preset
+        )
```

Comment:
I find it a _little_ odd that we initialize this twice, but it's probably fine, unless you have a different idea.